Add Helm & Operator instructions and cleanup Cluster Checks instructions

JacksonDavenport opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do?
<!-- A brief description of the change being made with this pull request.-->
- Add steps for setting up the Cluster Checks with Helm and the Operator. 
- Clean up several portions of the setup (DaemonSet steps for what to apply for files with ConfigMaps, don't need Unified Service Tagging section, re-arrange sections, clarify what the examples are)
- Provide more context to the Cluster Checks vs normal checks as well as what the Cluster Check Runners are

### Motivation
<!-- What inspired you to submit this pull request?-->
Add cleanup steps, and nudge users towards the easier ways of deployment (Helm & Operator)

### Preview
<!-- Impacted pages preview links-->
- https://docs-staging.datadoghq.com/jack.davenport/dca_checks/agent/cluster_agent/clusterchecks/?tab=helm

Current: 
- https://docs.datadoghq.com/agent/cluster_agent/clusterchecks/

### Additional Notes
<!-- Anything else we should know when reviewing?-->
Will repeat process for Endpoints Checks page in separate PR

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
```diff
@@ -11,48 +11,69 @@ further_reading:
-Then enable the cluster check feature:
+{{< tabs >}}
+{{% tab "Helm" %}}
+This is enabled by default in the Helm deployment of the Cluster Agent, with respect to the `datadog.clusterChecks.enabled` configuration key. The configuration can be set as:
```
```suggestion
Cluster check dispatching is enabled by default in the Helm deployment of the Cluster Agent through the `datadog.clusterChecks.enabled` configuration key:
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
````diff
@@ -11,48 +11,69 @@ further_reading:
+```
 
-Starting with version 1.2.0, the Datadog Cluster Agent extends the Autodiscovery mechanism for noncontainerized cluster resources. To enable this, make the following changes to the Cluster Agent deployment:
+This will enable the Cluster Check setup in the Cluster Agent and allow it to process configurations from the Kubernetes Service Annotations (`kube_services`).
````
```suggestion
This enables the cluster check setup in the Cluster Agent and allows it to process configurations from the Kubernetes Service Annotations (`kube_services`).
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
```diff
@@ -11,48 +11,69 @@ further_reading:
+This will enable the Cluster Check setup in the Cluster Agent and allow it to process configurations from the Kubernetes Service Annotations (`kube_services`).
+{{% /tab %}}
+{{% tab "Operator" %}}
+This can be enabled in the Operator deployment of the Cluster Agent with respect to the `clusterAgent.config.clusterChecksEnabled` configuration key. The configuration can be set as:
```
```suggestion
Cluster check dispatching is enabled in the Operator deployment of the Cluster Agent by using the `clusterAgent.config.clusterChecksEnabled` configuration key:
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
```diff
@@ -11,48 +11,69 @@ further_reading:
-2. Pass your cluster name as `DD_CLUSTER_NAME`. To help you scope your metrics, Datadog injects your cluster name as a `cluster_name` instance tag to all configurations.
-3. The recommended leader election lease duration is 15 seconds. Set it with the `DD_LEADER_LEASE_DURATION` environment variable.
-4. If the service name is different from the default `datadog-cluster-agent`, ensure the `DD_CLUSTER_AGENT_KUBERNETES_SERVICE_NAME` environment variable reflects the service name.
+This will enable the Cluster Check setup in the Cluster Agent and allow it to process configurations from the Kubernetes Service Annotations (`kube_services`).
```
```suggestion
This enables the cluster check setup in the Cluster Agent and allows it to process configurations from the Kubernetes Service Annotations (`kube_services`).
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
```diff
@@ -11,48 +11,69 @@ further_reading:
 Enable the `clusterchecks` configuration provider on the Datadog **Node** Agent. This can be done in two ways:
 
-- By setting the `DD_EXTRA_CONFIG_PROVIDERS` environment variable. This takes a space separated string if you have multiple values:
+- (Recommended) By setting the `DD_EXTRA_CONFIG_PROVIDERS` environment variable in your Agent DaemonSet. This takes a space separated string if you have multiple values:
```
```suggestion
- **Recommended**: By setting the `DD_EXTRA_CONFIG_PROVIDERS` environment variable in your Agent DaemonSet. This takes a space-separated string if you have multiple values:
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
````diff
@@ -66,30 +87,28 @@ Enable the `clusterchecks` configuration provider on the Datadog **Node** Agent.
     ```
 
-[Restart the Agent][1] to apply the configuration change.
+[1]: /agent/cluster_agent/setup/
````
Did you mean to delete the "Restart the Agent" line? if so, you can also remove the link URL (line 90)

JacksonDavenport replied:
Yes meant to remove that, that URL link is also referenced earlier too on line 65 so kept it for that.

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
```diff
@@ -66,30 +87,28 @@ Enable the `clusterchecks` configuration provider on the Datadog **Node** Agent.
-### Custom checks
-
-Running [custom Agent checks][5] as cluster checks is supported, as long as all node-based Agents are able to run it. This means your checks' code:
+**Note**: When running Cluster Checks the metrics reported by the Agent are not linked to a given hostname. As these are meant to be cluster centric metrics, and not necessarily host based metrics. As a result these metrics will also not inherit any host-level tags associated with that host. Such as those inherited from a cloud provider or added by the Agent's `DD_TAGS` environment variable. To a ...
```
```suggestion
**Note**: With cluster checks, the metrics reported by the Agent are not linked to a given hostname because they are meant to be cluster-centric metrics and not necessarily host-based metrics. As a result, these metrics do not inherit any host-level tags associated with that host, such as those inherited from a cloud provider or added by the Agent's `DD_TAGS` environment variable. To add tags to cluster check metrics, use the `DD_CLUSTER_CHECKS_EXTRA_TAGS` environment variable.
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
```diff
@@ -66,30 +87,28 @@ Enable the `clusterchecks` configuration provider on the Datadog **Node** Agent.
 
-- Must be installed on all node-based Agents where the `clusterchecks` config provider is enabled.
-- Must **not** depend on local resources that are not accessible to all Agents.
+The [Datadog Helm Chart][3] and the [Datadog Operator][4] additionally offer the possibility to deploy Cluster Check Runners. Which are a Deployment for a small set of Datadog Agents configured to run these dispatched cluster checks only, as opposed to dispatching these to the normal node-based Agents.
```
```suggestion
The [Datadog Helm Chart][3] and the [Datadog Operator][4] additionally offer the possibility to deploy cluster check runners, which are a deployment for a small set of Datadog Agents configured to run these dispatched cluster checks only, as opposed to dispatching these to the normal node-based Agents.
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
```diff
@@ -98,55 +117,114 @@ The following environment variables are required to configure the cluster check
 
-### Static configurations in files
+### Configuration from static configuration files
+When the URL or IP of a given resource is constant (eg. external service endpoint, public URL, etc.), a static configuration can be passed to the Cluster Agent as YAML files. The file name convention and syntax are the same as the static configurations on the node-based Agent. With the **required** addition of the `cluster_check: true` line.
```
```suggestion
When the URL or IP of a given resource is constant (for example, an external service endpoint or a public URL), a static configuration can be passed to the Cluster Agent as YAML files. The file name convention and syntax are the same as the static configurations on the node-based Agent, with the **required** addition of the `cluster_check: true` line.
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
```diff
@@ -98,55 +117,114 @@ The following environment variables are required to configure the cluster check
+
+{{< tabs >}}
+{{% tab "Helm" %}}
+With Helm these configuration files can be created within the `clusterAgent.confd` section. **Note**: This is separate from the `datadog.confd` section where the files are created in the node-based Agents. The `<INTEGRATION_NAME>` must exactly match the desired integration check you want to run.
```
```suggestion
With Helm, these configuration files can be created within the `clusterAgent.confd` section. **Note**: This is separate from the `datadog.confd` section where the files are created in the node-based Agents. The `<INTEGRATION_NAME>` must exactly match the desired integration check you want to run.
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
````diff
@@ -98,55 +117,114 @@ The following environment variables are required to configure the cluster check
+```
+{{% /tab %}}
+{{% tab "Daemonset" %}}
+With the manual approach you must create a `ConfigMap` storing the desired static configuration files and mount that into the corresponding `/conf.d` file of the Cluster Agent container. This follows the same approach for [mounting ConfigMaps into the Agent container][1]. For example:
````
```suggestion
With the manual approach you must create a `ConfigMap` to store the desired static configuration files, and mount it into the corresponding `/conf.d` file of the Cluster Agent container. This follows the same approach for [mounting ConfigMaps into the Agent container][1]. For example:
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
````diff
@@ -98,55 +117,114 @@ The following environment variables are required to configure the cluster check
+                path: <INTEGRATION_NAME>.yaml
+          #(...)
+```
+This should result in a file in the `/conf.d/` directory of the Cluster Agent with respect to your desired integration. For example: `/conf.d/mysql.yaml` or `/conf.d/http_check.yaml`.
````
```suggestion
This creates a file in the `/conf.d/` directory of the Cluster Agent corresponding to the integration. For example: `/conf.d/mysql.yaml` or `/conf.d/http_check.yaml`.
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
```diff
@@ -98,55 +117,114 @@ The following environment variables are required to configure the cluster check
+#### Example: MySQL check on an externally hosted database
 
-After setting up a CloudSQL instance and a [Datadog user][6], mount a `/conf.d/mysql.yaml` file in the Cluster Agent container with the following content:
+After setting up an externally hosted database, such as CloudSQL or RDS, and a corresponding [Datadog user][6] to access the database, mount a `/conf.d/mysql.yaml` file in the Cluster Agent container with the following content:
```
```suggestion
After you set up an externally hosted database, such as CloudSQL or RDS, and a corresponding [Datadog user][6] to access the database, mount a `/conf.d/mysql.yaml` file in the Cluster Agent container with the following content:
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
````diff
@@ -98,55 +117,114 @@ The following environment variables are required to configure the cluster check
 ```
 
-The `%%host%%` [template variable][7] is supported and is replaced by the service's IP. The `kube_namespace` and `kube_service` tags are automatically added to the instance.
+### Configuration from kubernetes service annotations
````
```suggestion
### Configuration from Kubernetes service annotations
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
```diff
@@ -177,15 +255,15 @@ spec:
 ### Kubernetes: find the leader Cluster Agent
 
-When leader election is enabled, only the leader serves Cluster Check configurations to the node-based Agents. The name of the leader is available in the `datadog-leader-election` ConfigMap:
+When leader election is enabled, only the leader serves Cluster Check configurations to the node-based Agents. If you only have 1 replica of the Cluster Agent pod running, that will be the leader. Otherwise, you can identify the name of the leader in the `datadog-leader-election` `ConfigMap`:
```
```suggestion
When leader election is enabled, only the leader serves Cluster Check configurations to the node-based Agents. If only one replica of the Cluster Agent pod is running, it is the leader. Otherwise, you can identify the name of the leader in the `datadog-leader-election` `ConfigMap`:
```

kayayarai commented on content/en/agent/cluster_agent/clusterchecks.md:
```diff
@@ -66,30 +87,28 @@ Enable the `clusterchecks` configuration provider on the Datadog **Node** Agent.
+#### Agent setup
 
-The following environment variables are required to configure the cluster check runners (or node Agents) to expose their check stats. The stats are consumed by the Cluster Agent and are used to optimize the cluster checks' dispatching logic.
+The following environment variables are required to configure the node Agents (or Cluster Check Runners) to expose their check stats. The stats are consumed by the Cluster Agent and are used to optimize the cluster checks' dispatching logic.
```
```suggestion
The following environment variables are required to configure the node Agents (or cluster check runners) to expose their check stats. The stats are consumed by the Cluster Agent and are used to optimize the cluster checks' dispatching logic.
```

kayayarai reviewed (changes requested):
Hi! Thank you for this excellent PR! I've made some edit comments. Please check that none of them change your intended meaning, and let me know if you have followup questions about them.

JacksonDavenport commented:
Thanks for the thorough review @kayayarai and for fixing that build failure. Was scratching my head with that. Added all your suggestions. 

Added to `Do Not Merge` label for now as just wanted to check in with Container Eng as well to see if they had any input here.

kayayarai commented:
Looks good @JacksonDavenport ! Ping me here or on Slack when you've had it reviewed and are ready to publish.

JacksonDavenport commented:
@kayayarai thanks! Should be fine, please merge whenever you get the chance.
